tests/kernel/mem_protect/syscalls/src/main.c

```diff
@@ -119,6 +119,45 @@ static inline int z_vrfy_syscall_arg64(uint64_t arg)
 }
 #include <syscalls/syscall_arg64_mrsh.c>
 
+ /* System call made for test test_system_call_separate_stack */
+int z_impl_syscall_separate_stack(void)
+{
+	/* Test that the variable is outside the thread's normal stack */
+	int syscall_var = 10;
+	int *var_ptr = &syscall_var;
+	intptr_t ptr_int = (intptr_t)var_ptr;
+	struct k_thread *thread = _current;
+	size_t size_addr = thread->stack_info.size;
+
+#if CONFIG_X86
+	unsigned int start_addr = thread->stack_info.start;
+
+	if ((ptr_int >= (start_addr + size_addr)) || (ptr_int <= start_addr)) {
+		return 0;
+	} else {
+		return -1;
+	}
+#elif defined(CONFIG_ARC) || defined(CONFIG_ARM)
+	unsigned int start_addr = thread->arch.priv_stack_start;
+
+	if ((ptr_int >= (start_addr + size_addr)) || (ptr_int <= start_addr)) {
+		return 0;
```

Comment:
I suppose, return 0 is the "success" case. Then you need to return -1 here, if the variable is outside the (start, start+size)